vochain/transaction/election_tx.go

```diff
@@ -43,9 +47,23 @@ func (t *TransactionHandler) NewProcessTxCheck(vtx *vochaintx.Tx) (*models.Proce
 	if !(tx.Process.GetStatus() == models.ProcessStatus_READY || tx.Process.GetStatus() == models.ProcessStatus_PAUSED) {
 		return nil, ethereum.Address{}, fmt.Errorf("status must be READY or PAUSED")
 	}
-	// check vtx.Signature available
-	if vtx.Signature == nil || tx == nil || vtx.SignedBody == nil {
-		return nil, ethereum.Address{}, fmt.Errorf("missing vtx.Signature or new process transaction")
+
+	// run specific checks based on census origin
+	switch tx.Process.CensusOrigin {
+	case models.CensusOrigin_OFF_CHAIN_CA:
+		if tx.Process.EnvelopeType.Anonymous {
+			return nil, ethereum.Address{}, fmt.Errorf("anonymous process not supported for CSP voting")
+		}
+	case models.CensusOrigin_FARCASTER_FRAME:
+		if tx.Process.EnvelopeType.Anonymous {
+			return nil, ethereum.Address{}, fmt.Errorf("anonymous process not supported for Farcaster voting")
+		}
+		if tx.Process.EnvelopeType.EncryptedVotes {
+			return nil, ethereum.Address{}, fmt.Errorf("encrypted votes not supported for Farcaster voting")
```

Comment:
```suggestion
			return nil, ethereum.Address{}, fmt.Errorf("encrypted votes not yet supported for Farcaster voting")
```

Reply:
Sorry to say, we cannot support secret until the end. The blockchain only receives buttonIndex=N as part of the proof and we need to verify buttonIndex == votePackage. So votePacakge cannot be encrypted.